crates/paimon/src/spec/types.rs

```diff
@@ -706,6 +922,55 @@ impl Default for TimeType {
     }
 }
 
+impl FromStr for TimeType {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if !s.starts_with(serde_utils::TIME::NAME) || s.contains("STAMP") {
```

Comment:
emm.. I feel here is a bit tricky.
BTW, Can `highly_complex_nested_row_type.json` pass this check ? We should also include this in tests ?